share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+```
+wake -x '"bad"| explode | drop 1| prepend "c" | cat' 
+```
+### Tuple
+A tuple is collection of items of different types. `get`,`set` and `edit` are the three inbuilt methods that can be used with the tuples.
+* Below example shows the declaration of a tuple. 
+```
+tuple Collection =
+  global IntValue : Integer
+  global DoubleValue : Double
+  global StringValue : String
+```
+
+`set<tuple name><field Name>` 
+`set` operation sets(writes) value to the tuple field
+`get<tuple name><field Name>` 
+`get` operation gets(reads) value to the tuple field
+`edit<tuple name><field Name>` 
+`edit` operation moidfies value to the tuple field
+
+### List
+A List is a sequence of items of same type. Empty list returns `Nil`. `head` returns the first element in the list. `tail` returns the rest of the list without the `head`. 
+* Example shows how to prepend the list of integers with the given value
+```
+def d = 1,2,3,Nil
+def listprepend data1 = prepend data1 d
+```
+
+* Example creates a list of given elements
+```
+def listcreate xx = xx,xx,xx,Nil
+```
+* Try out these examples
+```
+wake -x '"south" | explode | take 1| cat'
+wake -x '"north"| explode | tail | take 1 | cat'
+wake -x '"east" | explode| reverse | take 1 |cat'
+wake -x '"west" | explode | reverse | tail |take 1| cat'
+```
+
+* Below are the set of examples which shows how to extract specific characters from a string
+```
+def first s = s | explode | take 1| cat
+def second s = s | explode | tail | take 1 | cat
+def third s = s | explode| tail | tail | take 1 |cat
+def fourth s =  s | explode | tail | tail | tail | take 1| cat
+def last s = s| explode | reverse |take 1 | cat
+```
+
+* Try out these different set of examples and capture the result
+```
+def roll s = "{fourth s}{first s}{second s}{third s}"
+def exch s = "{second s}{first s}{third s}{fourth s}"
+```
+
+* Below example shows how to declare a list of strings and apply map function to it. Map is a predefined function that maps to the every element in the list
+```
+def words = "ache", "vile", "amid", "evil", "ogre" , Nil
+def mapping words = words | map roll
+def mapping2 words = map exch words
+```
+
+* Try out these examples and record the results
+```
+def what fullstring = fullstring| roll | roll | roll| exch |  roll
+def fb fullstring = fullstring | roll |roll | roll
+def fc fullstring = fullstring | roll | exch |fb
+def fd fullstring = fullstring | exch | fc | exch
+```
+
+* This examples convert the string "seat" -> "eats"
+```
+wake -x 'cat((tail (explode "seat")) ++(take 1(explode "seat")))'
+```
+
+### Type Inference
+* Here the paramater c has to be string since function strlen as been applied to it, and b has to be integer as it is being added and a also should be an integer because it is list of integers 
+```
+def madeup a b c =  (b + strlen c), a 
+```
+
+* Below example multiplies a given number by 6. The other way to implement this is `def times6 x = double x | triple`
```

Comment:
what does this have to do with type inference?

Reply:
I will move the Type Inference as a separate section to avoid the confusion